tests/installation/test_executor.py

```diff
@@ -1456,3 +1456,149 @@ def test_other_error(
     output = io.fetch_output().strip()
     assert output.startswith(expected_start)
     assert output.endswith(expected_end)
+
+
+def test_executor_known_hash_sha512(
```

Comment:
Is it necessary to have three similar test cases or is one parameterized test case sufficient?

Reply:
If you now add `packages_files` and `expected_url_reference` as parameters (like [that](https://github.com/python-poetry/poetry/blob/e31a9fc3ce5a7fc72eefa22040676bc66a35e5ad/tests/installation/test_executor.py#L247-L279)) that's what I meant. 😅